### ISSUE TITLE
release 0.5.0 DPP Bugs

### ISSUE THREAD

**Issue by onthebreeze** [uncefact/spec-untp#190]
### Impacted sections

 - https://uncefact.github.io/spec-untp/docs/specification/DigitalProductPassport 

### Issue Description

 - `Claim` object has both `declaredValue` and `declaredValues` property.  `declaredValues` should not be used.  It's a leftover from the transition from plural to singular properties 
 - `Product.traceabilityInformation` contains a single instance of `TraceabilityPerformance`.  It should be an array so that events can be grouped by lifecycle stage. 
 - `characteristics` placeholder for product specific properties is missing from the DPP model.

### Fix

To be implemented after pilot testing in release 0.6.0

**Comment by onthebreeze** [comment]
Also need to add a `name` property to `conformityAssessment` object of the Digital Conformity Credential

**Comment by absoludity** [comment]
> ### Issue Description
> 
>     * `Claim` object has both `declaredValue` and `declaredValues` property.  `declaredValues` should not be used.  It's a leftover from the transition from plural to singular properties

Yes, the property `Declaration.declaredValues` was renamed to the singular `declaredValue` way back as part of 0.4.1, but up until and including 0.5.0 both the DPP and DFR defined their own `Claim` entity inheriting from `core.Declaration` and explicitly listing the base fields. The DPP version in 0.5.0 hadn't been updated since the change and so the old plural `declaredValues` was additionally defined (the danger of explicitly naming fields from the base class, when the base class is updated).

![Image](https://github.com/user-attachments/assets/181a292d-d920-4b86-9c7d-30f9b2f67d56)

We noticed this a while back during the first alpha version of 0.6.0 ([documented here](https://github.com/uncefact/spec-untp/blob/65fcf3e5118629cd9ff0a8e9cadb626b5d5eb454/docs/untp-data-model/changes-v0.6.0-alpha1.md?plain=1#L207)).
So this was already fixed in the model as part of 0.6.0-alpha1.

The documentation link wasn't updated when 0.4.1 introduced the singular term. I've done that in https://github.com/uncefact/spec-untp/pull/353 .

> 
>     * `Product.traceabilityInformation` contains a single instance of `TraceabilityPerformance`.  It should be an array so that events can be grouped by lifecycle stage.

Done, that is, I've updated the model `ProductPassport.traceabilityInformation` (not `Product`) to be an array, verified the sample and saved a snapshot, so it will be so in the next release.

> 
>     * `characteristics` placeholder for product specific properties is missing from the DPP model.

The `Product.characteristics` is present on the `Product` entity in core:

![Image](https://github.com/user-attachments/assets/8519492d-b912-4761-9992-5c3a10772173)

But it is marked with the `[jsonld.contextOmit]=true` attribute so it won't be in the context (otherwise we get illegal redefinitions when it is defined for a specific extension).


> Also need to add a `name` property to `conformityAssessment` object of the Digital Conformity Credential

I've updated the DCC with `ConformityAssessment.name`. Similar to the above, it will be in the next release. Note: many `name` properties in core have `[jsonld.contextOmit]=true` and it's not clear to me why... it could be from when terms were all defined globally rather than within the context of the entity. I've not included it here (we'll see if there's an error before we release). Nor did I make the `ConformityAssessment.name` property required (let me know if it should be).

**Comment by colugo** [comment]
@absoludity I think `name` as a property name has issues with the `name` property from the VC context - or at least that's what I think I remember being told as the reason why it had to be omitted from the context.

**Comment by absoludity** [comment]
> [@absoludity](https://github.com/absoludity) I think `name` as a property name has issues with the `name` property from the VC context - or at least that's what I think I remember being told as the reason why it had to be omitted from the context.

Ah - yes, that's it:

https://github.com/w3c/vc-data-model/blob/f3732fdf27618a9b05b0f4e074920a4f6d09698e/contexts/credentials/v2#L20

Thanks @colugo . I'll add the `contextOmit` as well.

**Comment by absoludity** [comment]
All issues on this are addressed, so closing, even though the two changes haven't been released yet (they'll be included in the next release).
